update_versions.sh

```diff
@@ -1,18 +1,33 @@
-#! /usr/bin/env bash
+#!/bin/sh
 
+set -e
 
-ROOTDIR="$(cd "${0%/*}" 2>/dev/null || exit; pwd)"
-cd $ROOTDIR
-TEMP="$1.$$"
+[ -z "$1" ] && {
+    echo "Usage: $0 <FILE>, <FILE> is unspecified."
+    exit 1
+}
 
-{
-    echo "#include <string>"
-    echo "std::string pegof_version = \"$(git describe --tags --abbrev=8 --dirty)\";"
-    echo "std::string pcc_version = \"$(git -C packcc describe --tags --abbrev=8 --dirty)\";"
-} > "$TEMP"
+cd "${0%/*}"
 
-if [ ! -f "$1" ] || [ "$(md5sum < "$TEMP")" != "$(md5sum < "$1")" ]; then
-    mv "$TEMP" "$1"
-else
-    rm -f "$TEMP"
-fi
+A="$(git           describe --tags --abbrev=8 --dirty || git           rev-parse --short HEAD)"
+B="$(git -C packcc describe --tags --abbrev=8 --dirty || git -C packcc rev-parse --short HEAD)"
+
+f="$1.$$"
+
+cat > "$f" <<EOF
+#include <string>
+std::string pegof_version = "$A" ;
+std::string pcc_version = "$B" ;
+EOF
+
+[ -f "$1" ] && {
+    C="$(cat "$f")"
+    D="$(cat "$1")"
+
+    [ "$C" = "$D" ] && {
```

Comment:
Again, proper variable names would be better. Or you can avoid them completely:
```suggestion
    [ "$(cat "$f")" = "$(cat "$1")" ] && {
```